src/sentry/api/endpoints/group_hashes.py

```diff
@@ -69,3 +93,23 @@ def delete(self, request, group):
         )
 
         return Response(status=202)
+
+    def handle_results(self, results, project_id):
+        event_ids = map(lambda result: result['event_id'], results)
+        event_by_event_id = {
+            event.event_id: event
+            for event in Event.objects.filter(
```

Comment:
Can we get this from Snuba as well rather than going to the postgres table here?

Reply:
Unfortunately not. That was my initial goal too.

Snuba doesn't have some details like `metadata`, which is needed when displaying basic details about the latest event.

Reply:
We might be able to get get the 3 bits that are displayed here - title, message and location - out of Snuba instead of metadata eventually, but it might need some other changes first.

Reply:
"Snuba doesn't have some details like metadata, which is needed when displaying basic details about the latest event."

Not sure I follow. Are you saying you are missing metadata during serialization of Snuba events ?

Reply:
Snuba just doesn't have the parsed metadata that we need to display the compressed view of the event in the merged tab. So, need to query postgres to get those details.

Does that make it clearer? If not, I can show you in person.

Reply:
No, I don't think I get it still. We should not depend on postgres events anywhere since we are getting rid of that table. If the fields you need are "title, message and location" as Lynn said, then you should have them from the snuba dataset. Could you point me to the code that uses the parsed metadata ?

Reply:
Ah I see.

[Here](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx#L86)'s the call site and [here](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/events.tsx#L7-L22)'s the function using the parsed metadata.

Reply:
Ok that is not going to be a problem. If you load the event through the SnubaEvent class, it will populate the metadata from nodestore when you need it.

you can load a snuba event either one by one:
> SnubaEvent.get_event 
or
> SnubaEvent.objects.from_event_id

or you can laod snuba events multiple at a time like here:
https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/group_events.py#L122-L123